test/plugin/base_test.rb

```diff
@@ -1442,6 +1442,41 @@ def test_k8s_monitored_resource_fallback
     end
   end
 
+  # Test k8s monitored resource fallback when Metadata Agent restarts
+  # without the metadata server.
+  def test_k8s_monitored_resource_fallback_custom
```

Comment:
We could also refactor `test_k8s_monitored_resource_fallback` to pass in `config: <ACTUAL_CONFIG>` directly instead of `enable_metadata_agent: true/false` and then do:
```
        config: ENABLE_METADATA_AGENT_CONFIG,
        setup_metadata_agent_stub: false,
        setup_k8s_stub: false,
        log_entry: k8s_node_log_entry(log_entry(0)),
        expected_params: K8S_NODE_PARAMS
```
```
        config: CUSTOM_K8S_ENABLE_METADATA_AGENT_CONFIG,
        setup_metadata_agent_stub: false,
        setup_k8s_stub: false,
        log_entry: k8s_container_log_entry(log_entry(0)),
        expected_params: K8S_CONTAINER_PARAMS_CUSTOM
```
```
        config: CUSTOM_K8S_ENABLE_METADATA_AGENT_CONFIG,
        setup_metadata_agent_stub: false,
        setup_k8s_stub: false,
        log_entry: k8s_node_log_entry(log_entry(0)),
        expected_params: K8S_NODE_PARAMS_CUSTOM
```